legend-engine-ide-lsp-server/src/main/java/org/finos/legend/engine/ide/lsp/server/LegendLanguageService.java

```diff
@@ -229,4 +245,57 @@ public CompletableFuture<List<LegendEntity>> entities(LegendEntitiesRequest requ
             return entities;
         });
     }
+
+    @Override
+    public CompletableFuture<ApplyWorkspaceEditResponse> jsonEntitiesToPureTextWorkspaceEdits(LegendJsonToPureRequest request)
+    {
+        LegendSDLCFeature handler = this.server.getGlobalState().findFeatureThatImplements(LegendSDLCFeature.class).findAny().orElseThrow(() -> new RuntimeException("Could not find feature to convert json to pure files"));
+
+        return this.server.supplyPossiblyAsync(() ->
+        {
+            List<Either<TextDocumentEdit, ResourceOperation>> edits = new ArrayList<>();
+
+            for (String jsonFileUri : request.getJsonFileUris())
+            {
+                int lastIndexOf = jsonFileUri.indexOf("/src/main/legend/");
```

Comment:
It's confusing that the variable is named "lastIndexOf" but the method being called is indexOf rather than lastIndexOf.